### PR TITLE
Fix for wso2/product-iots#1035

### PR DESCRIPTION
**Pull request by harshanL**
This includes the fix and some minor refactoring work. With this fix the user will be prompted with a message 'Mail Server is not configured. Email invitation will not be sent' if **mailto** sender transport is not configured. Also there will be a warn log in the server log if it's not configured. As a result of this enhancement POST users/ response has been modified as below. **message** property will include both success and failure messages.

{
    "basicUserInfo" : {
              "username":"PRIMARY/username123",
              "firstname":"userfirstname",
              "lastname":"userlastname",
               "emailAddress":"user123@gmail.com" 
    },
    "message":"Mail Server is not configured. Email invitation will not be sent."
}

### CODE REVIEW

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/UserManagementServiceImpl.java [hunk 1]
```diff
@@ -563,7 +572,7 @@ public Response inviteExistingUsersToEnrollDevice(List<String> usernames) {
 
                 EmailMetaInfo metaInfo = new EmailMetaInfo(recipient, props);
                 dms.sendEnrolmentInvitation(DeviceManagementConstants.EmailAttributes.USER_ENROLLMENT_TEMPLATE,
-                        metaInfo);
+                                            metaInfo);
```
Comment (ruwany) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/UserManagementServiceImpl.java]: Formatting issue?

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementProviderServiceImpl.java [hunk 2]
```diff
@@ -55,6 +53,8 @@
 import org.wso2.carbon.device.mgt.common.operation.mgt.OperationManagementException;
 import org.wso2.carbon.device.mgt.common.operation.mgt.OperationManager;
 import org.wso2.carbon.device.mgt.common.policy.mgt.PolicyMonitoringManager;
+import org.wso2.carbon.device.mgt.common.pull.notification.PullNotificationExecutionFailedException;
```
Comment (ruwany) [inline components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementProviderServiceImpl.java:56]: Are these imports being used? I couldn't find references

**Review by ruwany — CHANGES_REQUESTED** [review]

File: components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementProviderServiceImpl.java (hunk 2 above)
Comment (harshanL) [reply components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementProviderServiceImpl.java:56]: This is used in https://github.com/wso2/carbon-device-mgt/blob/79abb6ef86c7895cc81679dfae5c85719f897761/components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/service/DeviceManagementProviderServiceImpl.java#L2308-L2308

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/UserManagementServiceImpl.java (hunk 1 above)
Comment (harshanL) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/UserManagementServiceImpl.java]: Thanks. Fixed.

**Review by ruwany — APPROVED** [review]

**Pull request merged**